.rhdh/docker/Dockerfile

```
@@ -2,7 +2,7 @@
 
 #@follow_tag(registry.redhat.io/rhel9/go-toolset:latest)
 # https://registry.access.redhat.com/ubi9/go-toolset
-FROM registry.access.redhat.com/ubi9/go-toolset:9.5-1747059472@sha256:a7d68b27e36edc6befe7dde94dd1db34c728a47df602998bd66c737d84813b5a AS builder
+FROM registry.access.redhat.com/ubi9/go-toolset:1.23@sha256:e0ad156b08e0b50ad509d79513e13e8a31f2812c66e9c48c98cea53420ec2bca AS builder
```

There's a strong chance that renovate will switch this to `9.6-1747333074` instead of 1.23 once it gets its claws into this repo, but this looks good to me. 

```suggestion
FROM registry.access.redhat.com/ubi9/go-toolset:9.6-1747333074@sha256:e0ad156b08e0b50ad509d79513e13e8a31f2812c66e9c48c98cea53420ec2bca AS builder
```